pkg/auth/providers/azure/azure_provider.go

```diff
@@ -68,114 +63,111 @@ func (ap *azureProvider) GetName() string {
-		SecurityEnabledOnly: &mem,
-	}
-
-	userGroups, err := azureClient.userClient.GetMemberGroups(context.Background(), *user.ObjectID, params)
+	userGroups, err := azureClient.ListGroupMemberships(clients.GetPrincipalID(userPrincipal))
 	if err != nil {
 		return nil, err
 	}
 
-	groupPrincipals, err := ap.userGroupsToPrincipals(azureClient, userGroups)
+	groupPrincipals, err := clients.UserGroupsToPrincipals(azureClient, userGroups)
 	if err != nil {
 		return nil, err
 	}
 
 	return groupPrincipals, nil
 }
 
-func (ap *azureProvider) SearchPrincipals(
-	name string,
-	principalType string,
-	token v3.Token,
-) ([]v3.Principal, error) {
-	var princ []v3.Principal
-
-	secret, err := ap.tokenMGR.GetSecret(token.UserID, Name, []*v3.Token{&token})
-	if err != nil && !apierrors.IsNotFound(err) {
+func (ap *azureProvider) SearchPrincipals(name string, principalType string, token v3.Token) ([]v3.Principal, error) {
+	cfg, err := ap.getAzureConfigK8s()
+	if err != nil {
 		return nil, err
 	}
+	var principals []v3.Principal
+	var secret string
+	if graphEndpointDeprecated(cfg) {
+		secret, err = ap.tokenMGR.GetSecret(token.UserID, Name, []*v3.Token{&token})
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, err
+		}
+	}
 
-	client, err := ap.newAzureClient(secret)
+	azureClient, err := ap.newAzureClient(cfg, secret)
 	if err != nil {
 		return nil, err
 	}
 
 	switch principalType {
 	case "user":
-		princ, err = ap.searchUsers(client, name, token)
+		principals, err = ap.searchUserPrincipalsByName(azureClient, name, token)
 		if err != nil {
 			return nil, err
 		}
 	case "group":
-		princ, err = ap.searchGroups(client, name, token)
+		principals, err = ap.searchGroupPrincipalsByName(azureClient, name, token)
 		if err != nil {
 			return nil, err
 		}
 	case "":
-		users, err := ap.searchUsers(client, name, token)
+		users, err := ap.searchUserPrincipalsByName(azureClient, name, token)
 		if err != nil {
 			return nil, err
 		}
-		groups, err := ap.searchGroups(client, name, token)
+		groups, err := ap.searchGroupPrincipalsByName(azureClient, name, token)
 		if err != nil {
 			return nil, err
 		}
-		princ = append(princ, users...)
-		princ = append(princ, groups...)
-
+		principals = append(principals, users...)
+		principals = append(principals, groups...)
 	}
-
-	return princ, ap.updateToken(client, &token)
+	return principals, ap.updateToken(azureClient, &token)
 }
 
-func (ap *azureProvider) GetPrincipal(
-	principalID string,
-	token v3.Token,
-) (v3.Principal, error) {
-	var princ v3.Principal
+func (ap *azureProvider) GetPrincipal(principalID string, token v3.Token) (v3.Principal, error) {
+	var principal v3.Principal
 	var err error
+	cfg, err := ap.getAzureConfigK8s()
+	if err != nil {
+		return v3.Principal{}, err
+	}
 
-	secret, err := ap.tokenMGR.GetSecret(token.UserID, Name, []*v3.Token{&token})
-	if err != nil && !apierrors.IsNotFound(err) {
-		return princ, err
+	var secret string
+	if graphEndpointDeprecated(cfg) {
```

Comment:
Nit:

You do this sort of "Is depracted/ if so fetch secret from token" call in the SearchPrincipals call too - might be worth just passing in the token to the "new client" method and having that function/method do this for us.